### PR TITLE
Ee 16843 monthly and tax year stats

### PR DESCRIPTION
**Pull request by owainrobertsbjss**
Adding new versions of generatePassRateStatistics which takes a TaxYear or a YearMonth as a parameter. Can be merged straight to master as it is still not wired up

### CODE REVIEW

File: src/test/java/uk/gov/digital/ho/proving/income/api/domain/TaxYearTest.java [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+
+    @Test
+    public void valueOf_null_illegalArgumentException() {
+        assertThatThrownBy(() -> TaxYear.valueOf(null))
+            .isInstanceOf(IllegalArgumentException.class)
+            .hasMessageContaining("from null");
+    }
+
+    @Test
+    public void valueOf_nonConsecutiveYears_illegalArgumentException() {
```
Comment (Tim-Hammonds) [inline src/test/java/uk/gov/digital/ho/proving/income/api/domain/TaxYearTest.java:87]: Could also have a test to show that the years must be in ascending order

File: src/main/java/uk/gov/digital/ho/proving/income/audit/statistics/PassRateStatisticsService.java [hunk 2]
```diff
@@ -35,6 +37,14 @@ public PassRateStatisticsService(AuditClient auditClient,
         this.requestPageSize = requestPageSize;
     }
 
+    public PassRateStatistics generatePassRateStatistics(YearMonth calendarMonth) {
+        return generatePassRateStatistics(calendarMonth.atDay(1), calendarMonth.atEndOfMonth());
+    }
+
+    public PassRateStatistics generatePassRateStatistics(TaxYear taxYear) {
```
Comment (Tim-Hammonds) [inline src/main/java/uk/gov/digital/ho/proving/income/audit/statistics/PassRateStatisticsService.java:44]: Have you got a set of tests for TaxYear args, in the same way there are for YearMonth args?

Comment (owainrobertsbjss) [reply src/main/java/uk/gov/digital/ho/proving/income/audit/statistics/PassRateStatisticsService.java:44]: Relies on the Tax Year to do calculation - testing interface to method

**Review by Tim-Hammonds — APPROVED** [review]

File: src/test/java/uk/gov/digital/ho/proving/income/api/domain/TaxYearTest.java (hunk 1 above)
Comment (owainrobertsbjss) [reply src/test/java/uk/gov/digital/ho/proving/income/api/domain/TaxYearTest.java:87]: Agreed - will address

**Pull request merged**